src/AbstractQuery.php

```diff
@@ -802,6 +802,24 @@ public function getSingleScalarResult(): mixed
         return $this->getSingleResult(self::HYDRATE_SINGLE_SCALAR);
     }
 
+    /**
+     * Gets the single scalar result of the query, returns null if result is empty.
+     *
+     * Alias for getSingleResult(HYDRATE_SINGLE_SCALAR) and try/catch NoResultException.
+     *
+     * @return bool|float|int|string|null The scalar result.
+     *
+     * @throws NonUniqueResultException If the query result is not unique.
+     */
+    public function getSingleScalarResultOrNull(): mixed
+    {
+        try {
+            return $this->getSingleResult(self::HYDRATE_SINGLE_SCALAR);
+        } catch (NoResultException) {
```

Comment:
We shouldn't use exception for flow control, they're expensive.

Reply:
@greg0ire what do you mean? `getOneOrNullResult` for examle already use `NoResultException`. Or you point is to delete `NoResultException` ?

Reply:
Oh right we already do this. But let's not add more code with the same bad practice. I mean, why is no result considered exceptional in the first place?

Reply:
But I also think you can completely remove it. Look at the code of `getSingleResult`.

```
        if ($this->_hydrationMode !== self::HYDRATE_SINGLE_SCALAR && ! $result) {
            throw new NoResultException();
        }
```

Reply:
Did you write a test the `catch`? I think that's impossible.

Reply:
After reading the thread at #7871 I am now confused as to what the right way would be. That `$hydrationMode` vs `$this->hydrationMode` thing that I didn't spot is hard to grasp.

Reply:
My bad, this behavior is actually proven by tests: https://github.com/doctrine/orm/blob/e7efdede15e572cf7950b0372391cac01a51152b/tests/Tests/ORM/Functional/QueryTest.php#L465-L470